onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -123,17 +123,31 @@ void UnaryOpQDQRules(SelectorActionRegistry& qdq_selector_action_registry) {
 void BinaryOpQDQRules(SelectorActionRegistry& qdq_selector_action_registry) {
   // 4 nodes. 2 x DQ for inputs, target, Q
   // Replace with internal QLinear version of operator. Delete all original nodes.
-  const std::string action_name{"2DQ"};
-  std::unique_ptr<Action> action = std::make_unique<QDQ::BinaryReplaceWithQLinear>(kMSDomain);
 
 #if !defined(ORT_MINIMAL_BUILD)
-  // TODO: Enable 16-bit types in selector when binary QLinear* ops support 16-bit.
-  std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::BinarySelector>();
-  qdq_selector_action_registry.RegisterSelectorAndAction(action_name,
-                                                         {{"Add", {}},
-                                                          {"Mul", {}}},
-                                                         std::move(selector),
-                                                         std::move(action));
+  {
+    const std::string action_name{"2DQ_Mul"};
+    std::unique_ptr<Action> action = std::make_unique<QDQ::BinaryReplaceWithQLinear>(kMSDomain);
+
+    std::vector<const char*> providers = {kCpuExecutionProvider};
+    std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::BinarySelector>(providers);
+    qdq_selector_action_registry.RegisterSelectorAndAction(action_name,
+                                                           {{"Mul", {}}},
+                                                           std::move(selector),
+                                                           std::move(action));
+  }
+
+  {
+    const std::string action_name{"2DQ_Add"};
+    std::unique_ptr<Action> action = std::make_unique<QDQ::BinaryReplaceWithQLinear>(kMSDomain);
+
+    std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::BinarySelector>();
+
+    qdq_selector_action_registry.RegisterSelectorAndAction(action_name,
+                                                            {{"Add", {}}},
+                                                            std::move(selector),
+                                                            std::move(action));
+  }
 
 #else
   qdq_selector_action_registry.RegisterAction(action_name, std::move(action));
```

Comment:
a name (and associated action) registered with `SelectorActionRegistry::RegisterSelectorAndAction()` in a full build must also be registered in a minimal build with `SelectorActionRegistry::RegisterAction()`. these names may be saved to an ORT format model so a minimal build needs to be able to handle them.